### PR TITLE
 Add creation of clusters tree

### PR DESCRIPTION
**Pull request by Roberta-Ferioli**
In this PR I add the output tree creation. I needed to change a little bit custer.h and clusterer.cxx to fix some problems with vectors. Now it runs, but we can't use Controller.py. It works with this list of commands (from src):  
.L ../../Utils/include/fileManager.cxx+
$ .L ../include/Cluster.h+
$ .L clusterer.cxx+
$ Clusterer c
$ c.Clustering("../data/input/DummyForTests3.txt")

### CODE REVIEW

**Review by GiorgioAlbertoLucia — CHANGES_REQUESTED** [review]
Dear @Roberta-Ferioli, thank you for this addition. There is currently no way to initialise the Cluster giving the clusterID to the constructor. Can you fix this as well?

**Comment by Roberta-Ferioli** [comment]
> Dear @Roberta-Ferioli, thank you for this addition. There is currently no way to initialise the Cluster giving the clusterID to the constructor. Can you fix this as well?

I didn't see that @fchinu had already implemented it in cluster.h. I will delete my change in the next commit

**Comment by Roberta-Ferioli** [comment]
@fchinu @GiorgioAlbertoLucia  I used clang-format in each file, but the test always fails. In my opinion is better delete this request. What do you think about it?

**Review by GiorgioAlbertoLucia — APPROVED** [review]
Thank you @Roberta-Ferioli for this important commit! It looks good

File: Clustering/data/input/DummyForTests3.txt [hunk 1]
```diff
@@ -1,20 +1,2 @@
...
-1 1 1 2 
-1 3 12 12
-1 3 13 12
-1 3 14 12
-2 5 123 123
-2 5 123 124
-2 5 122 124 
\ No newline at end of file
+event_count chip_id(decimal)
+hit_x hit_y 1 1 47 121 1 1 45 121 1 1 46 121 1 1 47 122 1 1 48 121 1 1 47 123 1 1 47 124 1 1 46 124 1 1 46 123 1 1 355 12 1 1 355 13 1 1 1 1 1 1 1 2 1 3 12 12 1 3 13 12 1 3 14 12 2 5 123 123 2 5 123 124 2 5 122 124 3 1 1 1 3 1 1 2 3 5 56 56
```
Comment (fchinu) [inline Clustering/data/input/DummyForTests3.txt]: Maybe better to revert this change

File: Clustering/include/Cluster_h_ACLiC_dict_rdict.pcm
Comment (fchinu) [inline Clustering/include/Cluster_h_ACLiC_dict_rdict.pcm:1]: I would remove this file and the .so one

File: .gitignore [hunk 2]
```diff
@@ -1,36 +1,32 @@
-# Prerequisites
+#Prerequisites
```
Comment (fchinu) [inline .gitignore]: I would remove the changes in this file, funny tho :)

File: Clustering/src/clusterer.cxx [hunk 3]
```diff
@@ -45,25 +80,26 @@ void Clusterer::Clustering(const char *inputFile) {
  * cluster is initialised with the hits found. After a cluster is found, it
  * moves on the following hit (not in the cluster) and repeats.
  */
-void Clusterer::EventClustering(FileManager &inputData, const int ievent) {
-
+std::vector<Cluster> Clusterer::EventClustering(FileManager &inputData,
```
Comment (fchinu) [inline Clustering/src/clusterer.cxx:83]: Is this really needed? You never call the Event Clustering to get the vector of clusters right?

File: Clustering/src/clusterer.cxx [hunk 4]
```diff
@@ -24,13 +32,40 @@ void Clusterer::Clustering(const char *inputFile) {
   FileManager inputData(inputFile);
 
   std::unordered_map<int, std::vector<int>> eventIndices;
+  // TTree * tree = new
+  // TTree(root["tree"]["clusterer"]["name"].As<std::string>().c_str(),// for
+  // the future if we will want a configuration file
+  // root["tree"]["clusterer"]["name"].As<std::string>().c_str());
+  TTree *tree = new TTree("Clusters_tree", "Clusters_tree");
+  std::vector<Cluster> clusterVector;
+  std::vector<std::vector<Cluster>> clustersVector;
```
Comment (fchinu) [inline Clustering/src/clusterer.cxx]: Just for future readability, I would call them EventClusters and RunClusters, or something like this to distinguish them

**Review by fchinu — CHANGES_REQUESTED** [review]
Many thanks @Roberta-Ferioli for these important changes! I just have few minor comments. Maybe if we compile instead of  just including files. If this doesn't fix the Controller.py I would move on with the project

File: Clustering/src/clusterer.cxx (hunk 3 above)
Comment (Roberta-Ferioli) [reply Clustering/src/clusterer.cxx:83]: you're right I forgot  to change it

File: Clustering/src/clusterer.cxx (hunk 4 above)
Comment (Roberta-Ferioli) [reply Clustering/src/clusterer.cxx]: ok

**Comment by Roberta-Ferioli** [comment]
too many problems, I opened another PR to avoid confluts
